src/data/finvec.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2020 Reid Barton. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Reid Barton
+-/
+
+import data.equiv.fin
+
+/-!
+# Finite vectors, implemented as functions on `fin n`.
+
+The main advantage of `finvec` over alternatives like `vector` is that
+extracting coordinates (components) of a `finvec` is just function application,
+making it easy to reason about things like permuting coordinates.
+
```

Comment:
Could you add a bit more text about what the alternatives are? My impression is, this is the "default" option for reasoning about spaces `X^n`. At least we use `fin n -> X` a lot already in mathlib. That could be noted too.

Reply:
`finvec` and `data/array` have some similarities

Reply:
Yeah, to be honest, I'm not really sure when you'd want to use alternatives like `vector`, so I don't know what else to say here.